src/DIRAC/Resources/Computing/ARCComputingElement.py

```diff
@@ -390,6 +394,10 @@ def killJob(self, jobIDList):
         for jobID in jobList:
             jobs.append(self._getARCJob(jobID))
 
+        # Set the token in the environment if needed
+        if self.token:
+            os.environ["BEARER_TOKEN"] = self.token["access_token"]
```

Comment:
I see a repeated pattern in most of these methods (common to at least ARC, ARC6, AREX), that is: 
1. call `_prepareProxy`, set it in `os.environ["X509_USER_PROXY"]`
2. add token to `os.environ["BEARER_TOKEN"]` if it exists

I would suggest to code a simple decorator that would do the above operations. It would reduce code repetition and have a clear entry pointy for future modifications.

Reply:
Agreed